File: android/Pets/app/src/main/java/com/pets/domain/favorites/FavoritesState.kt
package com.pets.domain.favorites

import com.pets.models.Pet
import com.pets.services.Interfaces.LogAction

data class FavoritesState(
    var list: List<Pet> = emptyList<Pet>(),
    var isLoading: Boolean = false
)

sealed class FavoritesEvent: LogAction
object FetchList: FavoritesEvent() {
    override fun description(): String {
        return "Action - Fetching Favorites List"
    }
}

class FetchListCompleted(val list: List<Pet>): FavoritesEvent() {
    override fun description(): String {
        return "Action - Fetching Favorites List Completed"
    }
}

class RemoveFromFavorites(val id: Int): FavoritesEvent() {
    override fun description(): String {
        return "Action - Removing from Favorites List"
    }
}

File: ios/Pets/Domain/Sources/Domain/Favorites/States/FavoritesState.swift
//
//  File.swift
//  
//
//  Created by Dylan Velez on 10/08/21.
//

import Foundation
import Models
import Services

public struct FavoritesState {
    public var list: Pets = []
    public var isLoading: Bool = false
    public init() { }
}

public enum FavoritesEvent {
    case fetchList
    case fetchListCompleted(pets: Pets)
    case removeFromFavorites(id: Int)
}

extension FavoritesEvent: LoggingAction {
    public var logDescription: String {
        switch self {
        case .fetchList:
            return "Action - Fetching favorites list"
        case .fetchListCompleted:
            return "Action - Completed Fetching favorites"
        case .removeFromFavorites:
            return "Action - Removed item from favorites"
        }
    }
}
